tests/SpecTests/ClientSideEncryptionSpecTest.php

```diff
@@ -1642,6 +1642,48 @@ static function (self $test, Client $setupClient, ClientEncryption $clientEncryp
         ];
     }
 
+    /**
+     * Prose test 15: On-demand AWS Credentials
+     *
+     * @see https://github.com/mongodb/specifications/tree/master/source/client-side-encryption/tests#on-demand-aws-credentials
+     * @testWith [true]
+     *           [false]
```

Comment:
The `$shouldSucceed=false` case corresponds to testOnDemandAwsCredentials_with_data_set__1 passes in https://spruce.mongodb.com/task/mongo_php_library_test_csfle_without_aws_credentials__php_edge_versions~latest_stable_mongodb_versions~6.0_driver_versions~latest_stable_os~debian11_test_without_aws_credentials_patch_bf91db2558c491f474c986bc50383ff002ec9e01_63857b60d6d80a169794c4e0_22_11_29_03_26_36/tests?execution=0&sortBy=STATUS&sortDir=ASC